airflow-core/src/airflow/ui/src/pages/Events/EventsFilters.tsx

```diff
@@ -0,0 +1,226 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, HStack, Input } from "@chakra-ui/react";
+import { useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { useSearchParams } from "react-router-dom";
+
+import { useTableURLState } from "src/components/DataTable/useTableUrlState";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+import { ResetButton } from "./ResetButton";
+import { getFilterCount, formatDateTimeLocalValue } from "./filterUtils";
+
+const {
+  AFTER: AFTER_PARAM,
+  BEFORE: BEFORE_PARAM,
+  DAG_ID: DAG_ID_PARAM,
+  EVENT_TYPE: EVENT_TYPE_PARAM,
+  MAP_INDEX: MAP_INDEX_PARAM,
+  RUN_ID: RUN_ID_PARAM,
+  TASK_ID: TASK_ID_PARAM,
+  TRY_NUMBER: TRY_NUMBER_PARAM,
+  USER: USER_PARAM,
+} = SearchParamsKeys;
+
+export const EventsFilters = () => {
+  const { t: translate } = useTranslation(["browse", "common"]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { setTableURLState, tableURLState } = useTableURLState();
+
+  const { pagination, sorting } = tableURLState;
+
+  const resetPagination = useCallback(() => {
+    setTableURLState({
+      pagination: { ...pagination, pageIndex: 0 },
+      sorting,
+    });
+  }, [pagination, setTableURLState, sorting]);
+
+  // Extract filter values from URL
+  const afterFilter = searchParams.get(AFTER_PARAM) ?? "";
+  const beforeFilter = searchParams.get(BEFORE_PARAM) ?? "";
+  const dagIdFilter = searchParams.get(DAG_ID_PARAM) ?? "";
+  const eventTypeFilter = searchParams.get(EVENT_TYPE_PARAM) ?? "";
+  const mapIndexFilter = searchParams.get(MAP_INDEX_PARAM) ?? "";
+  const runIdFilter = searchParams.get(RUN_ID_PARAM) ?? "";
+  const taskIdFilter = searchParams.get(TASK_ID_PARAM) ?? "";
+  const tryNumberFilter = searchParams.get(TRY_NUMBER_PARAM) ?? "";
+  const userFilter = searchParams.get(USER_PARAM) ?? "";
```

Comment:
```suggestion
  const afterFilter = searchParams.get(AFTER_PARAM);
  const beforeFilter = searchParams.get(BEFORE_PARAM);
  const dagIdFilter = searchParams.get(DAG_ID_PARAM);
  const eventTypeFilter = searchParams.get(EVENT_TYPE_PARAM);
  const mapIndexFilter = searchParams.get(MAP_INDEX_PARAM);
  const runIdFilter = searchParams.get(RUN_ID_PARAM);
  const taskIdFilter = searchParams.get(TASK_ID_PARAM);
  const tryNumberFilter = searchParams.get(TRY_NUMBER_PARAM);
  const userFilter = searchParams.get(USER_PARAM);
```

It's much easier to have `null` there and know that `null` mean the value was not fill, instead of having `""` everywhere in the code, not knowing if this was a user input or an empty filter.


You can take a look at `DagsFilters`

Reply:
Fixed